rmw_dds_common/test/test_graph_cache.cpp

```diff
@@ -1550,3 +1551,64 @@ TEST(test_graph_cache, bad_arguments)
     rcutils_reset_error();
   }
 }
+
+TEST(test_graph_cache, get_writers_info_by_topic_maybe_fail)
+{
+  RCUTILS_FAULT_INJECTION_TEST(
+  {
+    GraphCache graph_cache;
+    add_participants(graph_cache, {"participant1"});
+    add_entities(
+      graph_cache,
+    {
+      // topic1
+      {"reader1", "participant1", "topic1", "Str", true},
+      {"writer1", "participant1", "topic1", "Str", false},
+    });
+    add_nodes(
+      graph_cache, {
+      {"participant1", "ns1", "node1"},
+      {"participant1", "ns1", "node2"},
+      {"participant1", "ns2", "node1"}});
```

Comment:
@brawner perhaps not an issue, but should the test fixture be setup outside the fault injection loop?

Reply:
Yah, created a test fixture class.